packages/world/src/modules/core/implementations/BalanceTransferSystem.sol

```diff
@@ -37,6 +38,9 @@ contract BalanceTransferSystem is System, IWorldErrors {
       revert World_InvalidResourceType(RESOURCE_NAMESPACE, toNamespaceId, toNamespaceId.toString());
     }
 
+    // Require the namespace to exist
+    if (!ResourceIds.getExists(toNamespaceId)) revert World_ResourceNotFound(toNamespaceId, toNamespaceId.toString());
+
     // Require caller to have access to the namespace
     AccessControl.requireAccess(fromNamespaceId, _msgSender());
```

Comment:
I wonder how much gas we'd incur if we added this check in `requireAccess` instead?